src/control/security/grpc_authorization.go

```diff
@@ -77,6 +83,16 @@ var methodAuthorizations = map[string][]Component{
 	"/RaftTransport/InstallSnapshot":       {ComponentServer},
 }
 
+// MethodToComponent resolves a gRPC method string to a build.Component.
+func MethodToComponent(method string) (build.Component, error) {
+	comps, found := methodAuthorizations[method]
+	if !found || len(comps) == 0 {
```

Comment:
Could it be possible that len(comps) > 1

Reply:
`len(comps) != 1` may be a better check. It is possible to have multiple components, in which case this method can't tell which one is correct.

Reply:
Yeah, agreed that this helper can be more strict now that the caller can and should explicitly set the calling component. Fixed, and added a unit test for the helper as well.